src/NHibernate/Engine/StatefulPersistenceContext.cs

```diff
@@ -1474,6 +1475,15 @@ void IDeserializationCallback.OnDeserialization(object sender)
 					throw new InvalidOperationException(me.Message, me);
 				}
 			}
+
+			// Reconnect the lazy property proxies
+			foreach (var p in entitiesByKey)
+			{
+				if (p.Value is IFieldInterceptorAccessor lazyPropertyProxy && lazyPropertyProxy.FieldInterceptor != null)
```

Comment:
`AbstractEntityPersister.AfterReassociate` does also this kind of things, but it is meant for detached entities and also handle dirtiness, which should left untouched in the session deserialization case.